### DISCUSSION TITLE
Getting Symmetry Operations for Equivalent Atoms

### DISCUSSION THREAD

**Discussion by ejmeitz** [spglib/spglib#481] [Q&A]
Hello,

I am trying to understand the output for a couple simple crystals. I gave spglib a 4x4x4 mono-atomic FCC supercell and was able to verify that spglib identifies the space-group, primitive and conventional cells correctly. It also naturally said all 256 atoms are equivalent since there's only one species type and FCC has a one atom primitive cell. However, the symmetry output has 12288 operations. I was wondering if there is a way to know which operations map between equivalent atoms without brute force searching the list (e.g. if I take the atom at [0.625, 0.875, 0.75 ] which index(s) in the symmetry operations will map that back to its equivalent atom). In this example the rotation matrices are basically pointless since the primitive atom is just 0,0,0 so a brute force search is not hard but I am also interested in other potentially harder cases.

Another question I had arose when testing a simple 2x2x2 mono-atomic diamond super cell. Again the primitive cell etc. look good, but the atoms identified as the primitive cell do not include the origin atom and I would prefer if they did. Is there some way to force the origin atom to be included in the primitive cell?

Thanks!

**Comment by lan496** [comment]
>  I was wondering if there is a way to know which operations map between equivalent atoms without brute force searching the list (e.g. if I take the atom at [0.625, 0.875, 0.75 ] which index(s) in the symmetry operations will map that back to its equivalent atom).

At least spglib does not implement such an algorithm. You may be interested in https://journals.aps.org/prb/abstract/10.1103/PhysRevB.77.224115

>  Is there some way to force the origin atom to be included in the primitive cell?

I think there is no way to specify sites in the primitive cell in spglib.
Moreover, when an input supercell is skewed, some symmetry operations may not be returned: spglib decides to return integer matrices for rotations and some rotation matrices with the primitive basis vectors may not be represented in integer matrices.
